Net Runner/AppDelegate.mm

```diff
@@ -34,11 +34,25 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     
     assert( sizeof(float_t) == 4 );
     
-    // Load model bundles
-    
-    NSString *modelsPath = [[NSBundle mainBundle] pathForResource:@"models" ofType:nil];
+    // Move model bundles into Documents/ directory and then load them
```

Comment:
This is all the right way to do it. `URLsForDirectory` and grabbing the zeroth index is correct, as odd as it looks. And going back and forth between URLs and String paths is sometimes necessary if annoying.

You could change:

1. moving the calls into separate methods on the AppDelegate class to clean it up
2. then you can also drop variables that are only used once and embed those calls in the other calls
3. use `URLByAppendingPathComponent` instead of getting the `path` and then calling `stringByAppendingPathComponent` on that.

It's stupid that `NSFileManager` doesn't have a `fileExistsAtURL` method, but you could also use the `fileExistsAtPath:` version without checking if it's a directory, since you don't actually check the value of `checkDirectory` anyway :)

Something like:

```objc
- (NSString*)appModelsPath {
  return [[NSBundle mainBundle] pathForResource:@"models" ofType:nil];
}

- (NSString*)documentsModelsPath {
  NSURL *documentDirectoryURL = [[NSFileManager defaultManager] URLsForDirectory:NSDocumentDirectory inDomains: NSUserDomainMask][0];
  return [[documentDirectoryURL URLByAppendingPathComponent:@"models"] path];
}

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
  // ...

  NSFileManager *fileManager = [NSFileManager defaultManager];

  if ( ![fileManager fileExistsAtPath:[self documentsModelsPath] ) {
    NSError *error;
    if ( ![fileManager copyItemAtPath:[self appModelsPath] toPath:[self documentsModelsPath] error:&error] ) {
      NSLog(@"Error copying models: %@", error);
    }
  }
}
```

It ends up making some of the calls more than once, but the user is never going to notice the nanoseconds worth of difference, and it's more readable.